支持使用 MongoDB Atlas 作为向量数据库

Hoshino-Yumetsuki opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Added MongoDB support for vector store configuration.
	- Expanded vector store options to include MongoDB integration.
	- Implemented dynamic MongoDB client importing and vector store registration.

- **Documentation**
	- Updated configuration instructions to include MongoDB setup guidelines.
	- Added new entries for MongoDB settings in English and Chinese localization files.

- **Improvements**
	- Enhanced configuration flexibility for vector store management.
	- Added support for MongoDB Atlas vector search capabilities.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Hoshino-Yumetsuki committed:
refactor(long-memory): replace jieba with jieba-wasm for improved text segmentation

- Updated the dependency from `@node-rs/jieba` to `jieba-wasm` in package.json.
- Refactored the text segmentation logic in `similarity.ts` to utilize the new `cut` function from `jieba-wasm`, enhancing compatibility and performance.

Hoshino-Yumetsuki committed:
Merge branch 'ChatLunaLab:v1-dev' into v1-dev

Hoshino-Yumetsuki committed:
refactor(long-memory): enhance BM25 similarity calculation in similarity.ts

- Improved the BM25 similarity calculation by introducing term frequency maps for both documents.
- Added a smoothing factor and adjusted the scoring formula to normalize against the theoretical maximum score.
- Enhanced code readability and maintainability by restructuring the logic for term frequency and IDF calculations.

Hoshino-Yumetsuki committed:
style(long-memory): prettier

Hoshino-Yumetsuki committed:
Merge branch 'v1-dev' of https://github.com/Hoshino-Yumetsuki/chatluna into v1-dev

dingyi222666 commented on packages/vector-store-service/src/vectorstore/faiss.ts:
此处不需要，请更新主插件到最新 beta 版本

dingyi222666 commented on packages/vector-store-service/src/vectorstore/milvus.ts:
```diff
@@ -270,7 +287,7 @@ export async function apply(
                     vectorStore = undefined
                 }
-            }
+            } as ExtendedVectorStoreInput<Milvus>
```
不需要。

Hoshino-Yumetsuki committed:
feat(vector-store): add MongoDB configuration options to the vector store service

- Introduced new MongoDB configuration parameters: mongodbUrl, mongodbDbName, and mongodbCollectionName.
- Updated the configuration schema to include MongoDB as a supported vector store option.
- Added documentation link for MongoDB configuration in the usage section.

dingyi222666 commented on packages/vector-store-service/src/index.ts:
```diff
@@ -52,6 +57,14 @@ export const Config: Schema<Config> = Schema.intersect([
             .default('http://127.0.0.1:19530'),
         milvusUsername: Schema.string().default(''),
         milvusPassword: Schema.string().role('secret').default('')
+    }),
```
请在 locales 里添加 i18n 相关文本

dingyi222666 committed:
feat(vector-store): add MongoDB database settings to localization files
